src/net/rpc.rs

```diff
@@ -1132,6 +1277,13 @@ impl ConversationHttp {
             }
         };
 
+        if let Some(attachment) = attachment {
```

Comment:
What stops me from spamming a node to death with a bunch of garbage attachments for which there will never be a contract event?  Is there some kind of throttling logic or garage-collection logic somewhere?

Reply:
At this point, the only protection against this kind of attack is coming from the fact that the attachment registrations is happening after the mempool admission, meaning it would cost you STX to spam a node. 
I could be wrong, but if you have access to a node that you want to DDOS, flooding with txns would probably be easier, I remember the mempool admission being an expensive process.

But yeah we should think about garbage collecting, will track this here https://github.com/blockstack/stacks-blockchain/issues/2135